### DISCUSSION TITLE
Dynamically generate mock data for every operation

### DISCUSSION THREAD

**Discussion by zakariaelas** [orval-labs/orval#361] [Q&A]
Hello!

First, thanks to the author and contributors for the great library 💜 ! 

I have a mock data factory method that returns some dummy data depending on the tag provided. For example, my factory is capable of doing:
```javascript
// one product
const productMock = mockDataFactory.get("product")

// or many
const productsMock = mockDataFactory.list("product", 10)
```

I believe that the library supports overriding mock data, however, I think that the only way to do this is to override every single operation in `orval-config.js` like such:

```javascript
module.exports = {
  output: {
    // ... other config
    override: {
      operations: {
        GetProduct: {
          mock: {
            data: {..}
          },
        },
        ListProducts: {
          mock: {
            data: { ... }
          }
        }
        //... all other operations
    },
  },
}
```
Is there some way that I can use my mock data factory in order to provide mock data for every operation, without having to go through every operation?

I tried to leverage a transformer function like this one:
```javascript

function transformer(options) {
  const [tag] = options.tags
  const data = mockDataFactory.get(tag) // or list 
  return {
    ...options,
    override: {
      mock: {
        data,
      },
    },
  }
}
```

But unfortunately, the above does not work. It seems like the override field of the return verbOptions is not really used (or at least not for a similar purpose).

Any suggestions on how to make this possible?

**Comment by anymaniax** [comment]
Hello @zakariaelas! At the moment, you cannot do that but it's a good use case. I will think of a way to add this to the library. The only way is like you suggested just edit every operationId data property.

**Reply by zakariaelas** [reply]
Thanks for the reply @anymaniax ! Let me know if you want me to create an issue so it's easier to track. Can also try brainstorming some ideas if you're interested. Thanks again!
